### DISCUSSION TITLE
Migração/front -> back

### DISCUSSION THREAD

**Discussion by CinthiaQueiro** [frontendbr/forum#792] [Discussão]
Bom dia!

Sei que a comunidade aqui é de front, porém a minha dúvida é um pouco fora da área. Gostaria de saber se existe algum front aqui que tenha migrado da área de Back End. Estou querendo migrar para Back end e gostaria de dicas sobre o mercado... se possível algum canal no GitHub onde as coisas estejam mais "centralizadas" diríamos, como o frontbr... peço desculpas por lançar aqui a dúvida, mas se puderem ajudar agradeço.

**Comment by phenriqueleao** [comment]
[Aqui](https://github.com/backend-br/forum/issues)

**Comment by Cleysonlb** [comment]
@CinthiaQueiro eu iniciei na empresa que estou como Frontend, depois fui migrando para backend Ruby on Rails, pra mim a mudança foi bem natural, claro que com algumas dificuldades que é normal acontecer.
Te dou a dica de pegar uma tecnologia, estudar e aplicar, tipo Ruby on rails, Django, PHP etc...
Entenda bem o padrão MVC (Que não é um bicho de 7 cabeças) e estude bastante banco de dados, SQL é uma boa pedida nesse inicio!
Não da pra masterizar algo só com teoria  e hoje tem muito curso online bom que te da uma boa noção aprendendo o básico! tipo o [Alura](https://www.alura.com.br/planos-cursos-online?gclid=Cj0KCQjw4eXPBRCtARIsADvOjY3oaIVIdsNvnaHLkrJJSFlARpV-KnXimTUS-lc-mgGBwik0mdZ10y0aAhRHEALw_wcB) ou o [Udemy](https://www.udemy.com/)

**Comment by CinthiaQueiro** [comment]
Bom dia Cleyson

Então, eu atuei 7 anos em uma empresa como front, porém também fazia
bastante apoio na parte do back end. Mas, na época atuava apenas com php e
SQL, não tinha essa pegada de Frameworks....

Eu estava tentando buscar um pouco do que está "pegando" mais no mercado,
tipo Java, Ruby, Php para decidir exatamente o que estudar. Mas estou meio
perdida. rs

E também não sei se pelo fato de ter um bom tempo na área de Front se
conseguiria nessa altura do campeonato começar do zero para migrar para
júnior no Back End.

Tem dicas de livros ou comunidades mais centralizadas?

Obrigada :)

Em 1 de novembro de 2017 11:18, Cleyson Leal Braga <notifications@github.com

**Comment by doug2k1** [comment]
Node.js é um caminho interessante pra fazer essa migração. Você tem a vantagem da linguagem, que já é familiar pra quem é front, e boa parte do ferramental. E pelo que eu vejo a demanda no mercado tá quente.

**Comment by AlbericoD** [comment]
Olha, nunca é tarde para mudarmos, o importante é filtrar nosso objetivos e, enxergar o melhor das pessoas  no meio do trajeto, apesar de eu não gostar dessas divisões de trabalho na tecnologia, sou desenvolvedor backend, você vai ter uma ligeira curva de aprendizado caso seja possível migrar para o Node, mas no backend (acredito que no frontend também) não se deve levantar bandeira para nenhuma tecnologia, caso tenha a oportunidade de reforçar seu conhecimento  base, reforce, pratique: lógica de programação, estrutura de dados, aprenda a aprender, tudo faz a mesma coisa, não existe nada de novo e revolucionário  desde 1995, nada que no C, JAVA e javascript não exista/existia antes , essas novas tecnologias, frameworks são todas amalgamas destas bases, acredito que, você não quer ser uma grande especialista/usadora de apenas algumas tecnologias/ferramentas, claro, tem o seu valor naquele momento do _hype_, mas e depois que o momento acabar?!.  Então, ter a capacidade de resolver problemas, ser curiosa (instale coisas, mas veja como aquilo é feito) e base, sem dúvidas vai ser ótimo para sua carreira, vai ser fácil aprender qualquer coisa em qualquer momento da sua vida, desejo todo sucesso pra ti! \o/
